src/main/kotlin/pt/up/fe/ni/website/backend/model/permissions/Permissions.kt

```diff
@@ -0,0 +1,46 @@
+package pt.up.fe.ni.website.backend.model.permissions
+
+import java.util.TreeSet
+
+class Permissions(
+    permissions: Collection<Permission> = emptyList()
+) : MutableSet<Permission> by TreeSet(permissions) {
+
+    companion object {
+        fun fromLong(encoded: Long): Permissions {
+            val decodedPermissions = Permission.values()
+                .filter { perm ->
+                    // To decide if the permission is present in the `encoded` Long,
+                    // we need to see if the bits for those permissions are set (equal to 1)
+                    // `encoded ushr n and 1L` returns the value of the `n`-th bit of `encoded`
+                    val encodedBit = encoded ushr perm.bit and 1L
+                    return@filter encodedBit == 1L
```

Comment:
```suggestion
                    encodedBit == 1L
```
This should be the same. Do you prefer the other version because it's explicit?

Reply:
Yes, I generally prefer the use of return instead of using the version without the return, unless it's a single line operation. I am open to changing that, if you'd like me to.

Reply:
It's fine this way :)